### PR TITLE
Add timezone and date utils needed for ORM.

### PR DESCRIPTION
**Pull request by lucatume**
Ticket: #115579, tracked in MR with https://central.tri.be/issues/116356

This PR adds some utility functions to build and validate DateTime and DateTimeZone objects.

### CODE REVIEW

**Comment by tr1b0t** [comment]
Closing this Pull Request because it is pull requested against the wrong branch. The Central ticket is for *Sprint 2*, but this was pull requested against *release/M18.15*.

File: src/Tribe/Repository.php [hunk 1]
```diff
@@ -1760,7 +1758,8 @@ public function join_clause( $join ) {
 	/**
 	 * {@inheritdoc}
 	 */
-	public function where_clause( $where ) {
+	public function
+	where_clause( $where ) {
```
Comment (borkweb) [inline src/Tribe/Repository.php]: Looks like a goat walked on your keyboard here.

Comment (lucatume) [reply src/Tribe/Repository.php]: totally 🐐 insinuation.

**Comment by lucatume** [comment]
Fixed code issue, added changelog entry and created MR 18.16.

**Pull request merged**